src/operator/optimizer_op-inl.h

```diff
@@ -632,6 +632,7 @@ struct AdamParam : public dmlc::Parameter<AdamParam> {
   float beta1;
   float beta2;
   float epsilon;
+  float rho;
```

Comment:
it looks like rho doesn't need to be in backend? You can do the shinking in frontend.

Reply:
Yes. But `rho` is also a hyper-parameter. it should be added into `struct AdamParam` like that of `lr`. `lr` is shrinking in frontend but also have a part in backend.